src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs

```diff
@@ -533,54 +532,39 @@ private void OnPreviewControlStateChanged(object sender, EventArgs e)
         }
 
         /// <summary>
-        /// If mouse is over node or preview control, then preview control is expanded.
-        /// </summary>
-        private void ExpandPreviewControl()
-        {
-            if ((IsMouseOver || PreviewControl.IsMouseOver || DynCmd.IsTestMode) && PreviewControl.IsCondensed)
-            {
-                PreviewControl.TransitionToState(PreviewControl.State.Expanded);
-            }
-        }
-
-        /// <summary>
         /// Sets ZIndex of node the maximum value.
         /// </summary>
         private void BringToFront()
         {
-            if (IsMouseOver || PreviewControl.IsMouseOver || DynCmd.IsTestMode)
-            {
-                if (NodeViewModel.StaticZIndex == Int32.MaxValue)
-                {
-                    PrepareZIndex();
-                }
-                var index = ++NodeViewModel.StaticZIndex;
+            if (!IsMouseOver && !PreviewControl.IsMouseOver && !DynCmd.IsTestMode) return;
 
-                oldZIndex = nodeWasClicked ? index : ViewModel.ZIndex;
-                ViewModel.ZIndex = index;
+            if (NodeViewModel.StaticZIndex == int.MaxValue)
+            {
+                PrepareZIndex();
             }
+
+            var index = ++NodeViewModel.StaticZIndex;
+
+            oldZIndex = nodeWasClicked ? index : ViewModel.ZIndex;
+            ViewModel.ZIndex = index;
         }
 
         private void OnPreviewControlMouseEnter(object sender, MouseEventArgs e)
         {
             if (PreviewControl.IsCondensed)
             {
-                Dispatcher.DelayInvoke(previewDelay, ExpandPreviewControl);
+                PreviewControl.TransitionToState(PreviewControl.State.Expanded);
             }
         }
 
         private void OnPreviewControlMouseLeave(object sender, MouseEventArgs e)
         {
-            if (!PreviewControl.StaysOpen && !PreviewControl.IsInTransition
-                && Keyboard.Modifiers != System.Windows.Input.ModifierKeys.Control
-                && !IsMouseOver
-                && Mouse.Captured != null && !IsMouseInsideNodeOrPreview(e.GetPosition(this)))
+            if (!PreviewControl.StaysOpen)
```

Comment:
these checks were for preview control mouse leave event and now this method is for pin icon. IsInTransition check was needed because of animations, the others were redundant as for me